internal/daemonset/daemonset.go

```diff
@@ -0,0 +1,216 @@
+package daemonset
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/common"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/identifiers"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/results"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+	"github.com/test-network-function/cnf-certification-test/pkg/tnf"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	daemonSetName    = "partner-repo"
+	namespace        = "default"
+	containerName    = "container-00"
+	imageWithVersion = "quay.io/testnetworkfunction/debug-partner:latest"
+	timeout          = 5 * time.Minute
+	nodeExporter     = "node-exporter"
+)
+
+var _ = ginkgo.Describe(common.Daemonset, func() {
+	var env provider.TestEnvironment
+
+	ginkgo.BeforeEach(func() {
+		env = provider.GetTestEnvironment()
+	})
+	ginkgo.ReportAfterEach(results.RecordResult)
+	ginkgo.AfterEach(func() {
+		env.SetNeedsRefresh()
+	})
+
+	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestICMPv4ConnectivityIdentifier)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		partnerRepoDaemonset()
+	})
+
+})
```

Comment:
I am not sure I understand this part.  This isn't a test, we are just trying to deploy the daemonset in the code versus the call to `install-partner-pods` [here](https://github.com/test-network-function/cnf-certification-test/blob/968feaa74a036b4893d6560114e8199817d371c0/run-cnf-suites.sh#L105).

I would think we would call to create the daemonset somewhere around [here](https://github.com/test-network-function/cnf-certification-test/blob/main/cnf-certification-test/suite_test.go#L162) in TestTest() that would deploy the daemonset first before the call to `ginkgo.RunSpecs(t, CnfCertificationTestSuiteName)`.